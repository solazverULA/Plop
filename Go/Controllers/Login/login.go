package login

import (
	/*"bytes"
	"crypto/tls"
	
	
	//"html/template"
	"io/ioutil"
	"log"*/
	"net/http"
	"fmt"
	"encoding/json"
	//"net/mail"
	//"net/smtp"
	"../../Controllers/User"
	"../../Models/ModelUser"
	//"github.com/gorilla/mux"
	/*"mime/multipart"
	"strconv"
	"github.com/gorilla/mux"
	"math/rand"
	"time"*/
)

//Iniciar sesion
func Login(w http.ResponseWriter, r *http.Request) {

	usersend := user.GetRequestUser(r)
	//fmt.Println("llego aqui")
	var status string
	var message string

	w.Header().Set("Content-Type", "text/html; charset-utf-8")
	w.Header().Set("Access-Control-Allow-Origin", "*")

	//conexion base de datos
	user := modeluser.GetUsuario(usersend.Email)
	fmt.Println("password", user)

	err := modeluser.ComparePassword(usersend.Password, user)

	if !modeluser.ValideEmail(usersend.Email, user) {
		status = "error"
		message = "Usuario no existe"
	} else {

		if err == nil {
			status = "success"
			message = "Bienvenido"
		} else {
			status = "error"
			message = "Usuario o contrase√±a incorrectos"
			//log.Fatal(err)
		}
	}

	response := modeluser.ResponseUser{status, user, message}

	//conexion con json
	json.NewEncoder(w).Encode(response)
}