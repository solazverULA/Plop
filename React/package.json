{
  "name": "nud-react",
  "version": "1.0.0",
  "private": true,
  "homepage": "https://timcreative.github.io/now-ui-dashboard-react",
  "dependencies": {
    "animate.css": "^3.6.1",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "bootstrap": "4.0.0-beta",
    "chart.js": "^2.7.2",
    "history": "4.7.2",
    "jquery": "3.3.1",
    "node-sass": "^4.9.0",
    "node-sass-chokidar": "^1.3.0",
    "npm-run-all": "4.1.1",
    "perfect-scrollbar": "1.3.0",
    "react": "16.1.0",
    "react-animated-css": "^1.0.4",
    "react-chartjs-2": "2.7.0",
    "react-dom": "16.1.0",
    "react-google-maps": "9.2.2",
    "react-notification-alert": "0.0.7",
    "react-router-dom": "4.2.2",
    "react-scripts": "^1.1.4",
    "reactstrap": "5.0.0-alpha.4"
  },
  "devDependencies": {
    "babel-jest": "^22.0.4",
    "jest": "^22.0.4",
    "react-test-renderer": "^16.0.0"
  },
  "scripts": {
    "build-css": "node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/",
    "watch-css": "npm run build-css && node-sass-chokidar --include-path ./src --include-path ./node_modules src/ -o src/ --watch --recursive",
    "start-js": "react-scripts start",
    "start": "npm-run-all -p watch-css start-js",
    "build": "npm run build-css && react-scripts build",
    "test": "jest",
    "jest-update-snapshot": "jest --updateSnapshot",
    "eject": "react-scripts eject"
  }
}
