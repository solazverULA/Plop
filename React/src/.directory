[Dolphin]
Timestamp=2018,6,27,3,57,52
Version=3
