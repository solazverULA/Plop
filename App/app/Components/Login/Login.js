import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TextInput,
  Button,
  AsyncStorage,
  Alert,
  Switch, 
  Linking,
  WebView,
  AlertIOS,
} from 'react-native';
import { FormLabel, FormInput,FormValidationMessage } from 'react-native-elements';
import Api from "../../Api/Api";
import Languaje from "../../Api/LanguajeApi"

import PhoneInput from 'react-native-phone-input'

export default class Login extends Component<{}> {
 
  constructor(props) {
    super(props);
  
    this.state = {Numberphone:"", errNumero:"", Err:false};
  }

  _onRegisterPhone(Numberphone){

    if(Numberphone==""){
      this.setState({Err:true, errNumero:"Introduce tu numero"})
    }else {

      Api._SubscribeListener(Numberphone, (data)=>{
         
        if(data.Data.Id!=0){
            Api._getLastNotificationListener(data.Data.Id, (notification)=>{
              if(notification.Id!=0){
                let url = (notification.Type == 2 || notification.Type == 3 || notification.Type == 6) ? notification.Action : "https://notificatorapp.com/Admin/#/notification/"+ notification.Srcimage;
                if (notification.Type == 7) 
                  url = "https://docs.google.com/document/d/"+notification.Srcimageexpandible+"/edit"
                if (notification.Type == 8)
                  url = "https://notificatorapp.com/Admin/#/information/"+ notification.Id
                if(!url.includes("https://") && !url.includes("http://")){
                  url= "http://" + url;
                }
                 window.location = url;
               
                Linking.openURL(url)
              }
                

              })
          AsyncStorage.setItem('ListenerData', JSON.stringify(data.Data))
            .then((data)=>{
             
              this.props._goHome();
            })
            .catch((err)=>Alert.alert("exitos iniciando session", JSON.stringify(err)))
        
        } else {
            if(Numberphone.length>7)
              this.setState({Err:true, errNumero:"Su numero no se encuentra registrado"})
        }
      })

      
    }
  }
  componentDidMount() { // B
    if (Platform.OS === 'android') {
      Linking.getInitialURL().then(url => {
       // alert( url)
        this.navigate(url);
      });
    } else {
        Linking.addEventListener('url', this.handleOpenURL);
    }
  }
  navigate = (url) => { // E
    
    const route = url.replace(/.*?:\/\//g, '');
   
    const id = route.match(/\/([^\/]+)\/?$/)[1];
    alert("https://notificatorapp.com/Admin/#/" + id + "/listener")
    this.setState({url:"https://notificatorapp.com/Admin/#/" + id + "/listener"})
    const routeName = route.split('/')[0];

    
  }
  /*
  <Text style={{fontSize: 22,  textAlign: 'center',margin: 10, fontWeight:'bold'}}>
          
        </Text>
       
        
        <FormLabel> NÃºmero de telefono</FormLabel>
        <View >
          <PhoneInput
                  ref='phone'
                  initialCountry='co'

                  onChangePhoneNumber={(Text)=>this._onRegisterPhone(String(Text))}

                  isValidNumber
                  textProps={{placeholder: 'Introduzca su numero aqui'}}
                  style={{width:"80%", height:30}}
          />
          <FormValidationMessage>{this.state.errNumero}</FormValidationMessage>
        </View>  
   */
  render() {
    return (
      <View style={styles.container}>
        <WebView
          source={{uri: this.state.url}}
          style={{marginTop: 20}}
        />     
        
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
