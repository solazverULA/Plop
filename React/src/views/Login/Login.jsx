import React from 'react';
import {
    Card, CardHeader, CardBody, CardFooter, CardTitle, Button, Row, Col, UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem, Table, Input,  InputGroup, FormGroup
} from 'reactstrap';
// react plugin used to create charts
import { Line, Bar } from 'react-chartjs-2';
// function that returns a color based on an interval of numbers

import {
    PanelHeader, Stats, CardCategory, Tasks
} from '../../components';

import {
    dashboardPanelChart,
    dashboardShippedProductsChart,
    dashboardAllProductsChart,
    dashboard24HoursPerformanceChart
} from '../../variables/charts.jsx';

import { tasks } from '../../variables/general.jsx';

import Api from '../../api/Api/Api'

class Login extends React.Component{

  loginUser() {
    Api._loginUser(this.state, (data)=>{
      if(data.Status === "success"){
        window.location.href="/Dashboard"
      }
      console.log(data)
    })
  }

    render(){
        return (
            <div>
                <PanelHeader
                    size="sm"
                />
              <div className="content">
                    <Row className="justify-content-center">
                        <Col xs={12} md={4} className="justify-content-center">
                            <Card className="card-chart" style={{height: '300px'}}>
                                <CardHeader style={{paddingTop:'7%'}}>
                                    <CardTitle>Login</CardTitle>
                                </CardHeader>
                                <CardBody>
                                    <div xs={10} md={2} lg={2} style={{paddingTop:'2%'}}>
                                      <FormGroup>
                                        <div>
                                          E-mail
                                        </div>
                                        <InputGroup>
                                          <Input onChange={(event)=>this.setState({Email:event.target.value})} type="email" id="login_email" name="Email" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                      <FormGroup>
                                        <div>
                                          Password
                                        </div>
                                        <InputGroup>
                                          <Input onChange={(event)=>this.setState({Password:event.target.value})} type="password" id="login_password" name="Password" required={true}/>
                                        </InputGroup>
                                      </FormGroup>
                                    </div>
                                </CardBody>
                                <CardFooter className='justify-content-right text-right' style={{marginTop:'-2%'}}>
                                  <Button onClick={this.loginUser.bind(this)} color="primaryBlue" style={{width:'100px'}}>login</Button>
                                </CardFooter>
                            </Card>
                        </Col>
                    </Row>
                </div>
            </div>
        );
    }
}

export default Login;
