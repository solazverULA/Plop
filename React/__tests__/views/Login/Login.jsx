import React from 'react';
import Login from '../../../src/views/Login/Login';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const loading = renderer.create(
    <Login/>
  ).toJSON();
  
  expect(loading).toMatchSnapshot();
});
