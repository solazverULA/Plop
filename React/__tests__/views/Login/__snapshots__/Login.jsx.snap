// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="panel-header panel-header-sm"
  />
  <div
    className="content"
  >
    <div
      className="justify-content-center row"
    >
      <div
        className="justify-content-center col-12 col-md-4"
      >
        <div
          className="card-chart card"
          style={
            Object {
              "height": "300px",
            }
          }
        >
          <div
            className="card-header"
            style={
              Object {
                "paddingTop": "7%",
              }
            }
          >
            <h4
              className="card-title"
            >
              Login
            </h4>
          </div>
          <div
            className="card-body"
          >
            <div
              lg={2}
              md={2}
              style={
                Object {
                  "paddingTop": "2%",
                }
              }
              xs={10}
            >
              <div
                className="form-group"
              >
                <div>
                  E-mail
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="login_email"
                    name="Email"
                    onChange={[Function]}
                    required={true}
                    type="email"
                  />
                </div>
              </div>
              <div
                className="form-group"
              >
                <div>
                  Password
                </div>
                <div
                  className="input-group"
                >
                  <input
                    className="form-control"
                    id="login_password"
                    name="Password"
                    onChange={[Function]}
                    required={true}
                    type="password"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="justify-content-right text-right card-footer"
            style={
              Object {
                "marginTop": "-2%",
              }
            }
          >
            <button
              className="btn btn-primaryBlue"
              onClick={[Function]}
              style={
                Object {
                  "width": "100px",
                }
              }
              type="button"
            >
              login
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
